vendor/brave-ios/Ledger/BATBraveLedger.mm

```diff
@@ -466,19 +466,20 @@ - (void)listRecurringTips:(void (NS_NOESCAPE ^)(NSArray<BATPublisherInfo *> *))c
 
 - (void)addRecurringTipToPublisherWithId:(NSString *)publisherId amount:(double)amount
 {
-  [BATLedgerDatabase insertOrUpdateRecurringTipWithPublisherID:publisherId
-                                                        amount:amount
-                                                     dateAdded:[[NSDate date] timeIntervalSince1970]
-                                                    completion:^(BOOL success) {
-                                                      if (!success) {
-                                                        return;
-                                                      }
-                                                      for (BATBraveLedgerObserver *observer in self.observers) {
-                                                        if (observer.recurringTipAdded) {
-                                                          observer.recurringTipAdded(publisherId);
-                                                        }
-                                                      }
-                                                    }];
+  ledger::ContributionInfoPtr info = ledger::ContributionInfo::New();
+  info->publisher = publisherId.UTF8String;
+  info->value = amount;
+  info->date = [[NSDate date] timeIntervalSince1970];
+  ledger->SaveRecurringTip(std::move(info), ^(ledger::Result result){
+    if (result != ledger::Result::LEDGER_OK) {
```

Comment:
completion block should be called above this line after adding it (seen in previous comment): 
```objc
dispatch_async(dispatch_get_main_queue(), ^{
  completion(result == ledger::Result::LEDGER_OK)
});
```

Reply:
edit: you don't need the `dispatch_async` to main queue, the database insertion callbacks are done on main queue already